core/Db/Settings.php

```diff
@@ -32,9 +33,14 @@ public function getUsedCharset()
         return strtolower($this->getDbSetting('charset'));
     }
 
+    public function getRowFormat()
+    {
+        return $this->getDbSetting('enable_dynamic_row_format') ? 'ROW_FORMAT=DYNAMIC' : '';
```

Comment:
@peterhashair I'm now actually wondering if we actually need this config setting? If I understand it now right, then we can apply that row format whenever `$this->getUsedCharset() === 'utf8mb4'`?  Is that correct @sgiehl  or would we still need a setting in case some DB doesn't support this row format?

Reply:
@tsteur  That's right, do you want me to remove the configuration, apply it to default if ut8mb4.

Reply:
In theory there might be cases where `ROW_FORMAT=DYNAMIC` might be supported, but `utf8mb4` collation is disabled manually. Not sure if anyone would do that or if that makes sense at all 🤷 
But guess for simplicity we could for now go without the config option and if there are coming too many complains that it breaks something we can introduce the config option later maybe?